en/developer_guide/bbcode.rst

```diff
@@ -0,0 +1,136 @@
+
+Add a new BBCode to the parser requires to add its definition. There are two ways of adding a
+definition.
+
+Create a simple BBCode tag via configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To create a simple BBCode definition, its possible to use a trivial configuration. The parameter
+``open_orchestra_bbcode.code_definitions`` contains an array of declared simple tags. Here is for
+instance the declaration of the bold tag:
+
+::
+
+    bold:
+        tag: b
+        html: <strong>{param}</strong>
+
+- The key 'bold' is the name of the definition used internally by the parser.
+
+- The ``html`` key is the html pattern of the code. The string ``{param}`` will be replaced by the
+content of the BBcode tag.
+
+- The ``tag`` key is the BBCode pattern of the tag. In this example the tag is [b]{param}[/b].
+
+Simple tags can also include some options. For instance the tag [h] which matches a <h*> html tag
+requires an option to indicate the level of the header. For instance [h=1]Some text[/h] will
+produce a <h1> html tag. You still can declare this in a configuration file like this:
+
+::
+
+        title:
+            tag: h
+            html: <h{option}>{param}</h{option}>
+            parameters:
+                use_option: true
+
+To declare that a tag uses option, you must add the key ``parameters.use_option`` and set it to true.
+Not that when declaring a tag with option, that option becomes mandatory. So if the option is not
+mandatory, you should declare two versions of the tag: one with option and one without. That way
+the parser will use one or the other according to the presence of an option. The ``html`` key uses
+both {param} and {option} string to indicate where to place those informations.
+The resulting BBCode will be [h={option}]{param}[/h]
+
+Create a simple BBCode tag via service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+BBCode definition can also be declared by services. A compiler pass has been added to the
+BBCodeBundle to collect all services tagged with ``open_orchestra_bbcode.code_definitions`` and add
+the matching definition to the master definition set.
+
+Such a service must implement ``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionCollectionInterface``.
+This collection must be composed of elements implementing
+``OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionInterface``. You can quickly generate
+definitions implementing that interface using the OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinitionFactory.
+This factory can be accessed via the service named ``open_orchestra_bbcode.definition.factory``
+
+To generate a new simple definition with this factory, call the method ``create`` with the following
+parameters:
+
+- $tag: the BBCode tag, ie ``b`` for the tag [b]
+
+- $html: the html representation of the tag, ie ``<strong>{param}</strong>`` for the [b]tag
+
+- $use_option: whether the tag uses option or not
+
+So to create a bold tag definition object, you can write:
+
+::
```

Comment:
see https://raw.githubusercontent.com/open-orchestra/open-orchestra-docs/master/en/developer_guide/custom_fixtures_load.rst for usage